python/equistore-operations/tests/same_raise.py

```diff
@@ -0,0 +1,87 @@
+"""Check that errors are raised for unequal meta data in several operations."""
+
+from os import path
+
+import numpy as np
+import pytest
+
+import equistore
+from equistore import NotEqualError
+
+
+DATA_ROOT = path.join(path.dirname(__file__), "data")
+
+
+def tensor():
+    tensor = equistore.load(
+        path.join(DATA_ROOT, "qm7-power-spectrum.npz"),
+        use_numpy=True,
+    )
+
+    # Ensure that the tensor has at least one gradient. This avoids dropping
+    # gradient tests silently by removing gradients from the reference data
+    assert "positions" in tensor.block(0).gradients_list()
```

Comment:
I don't understand, tensor is directly loaded from the file how can it have not a gradient if it is in the file? Is some test writing to the file?

Reply:
I feel like this assert makes in makes more more sense in the context of the test but I can also see that you don't want that this happens again in a new test. I am okay with this.

For me this is more understandable:
```suggestion
    assert len(tensor.block(0).gradients_list()) > 0, ("Tensor must have at least one gradient. When no gradients are present certain tests will pass without testing anything.")

```

Reply:
Yes, this is a more helpful message. I added this to the test.